libvast/src/experimental_table_slice_builder.cpp

```diff
@@ -665,14 +665,46 @@ std::shared_ptr<arrow::Schema> make_experimental_schema(const type& t) {
   std::vector<std::shared_ptr<arrow::Field>> arrow_fields;
   arrow_fields.reserve(rt.num_leaves());
   for (const auto& [field, index] : rt.leaves()) {
-    auto field_ptr
-      = arrow::field(rt.key(index), make_experimental_type(field.type));
-    arrow_fields.emplace_back(std::move(field_ptr));
+    arrow_fields.emplace_back(
+      make_experimental_field(rt.key(index), field.type));
   }
   auto metadata = arrow::key_value_metadata({{"name", std::string{t.name()}}});
   return std::make_shared<arrow::Schema>(arrow_fields, metadata);
 }
 
+std::shared_ptr<arrow::Field>
+make_experimental_field(const std::string& name, const type& t) {
+  using data_type_ptr = std::shared_ptr<arrow::Field>;
+  auto f = detail::overload{
+    [=](const auto& x) -> data_type_ptr {
```

Comment:
I think the auto overload is quite dangerous. Maybe we should only have this for basic types, e.g.,

```
[&]<basic_type T>(const T& x) -> data_type_ptr { ... }
```

Then the compiler would error for new complex types that an overload is missing instead of silently using this one.